purchase_sale_stock_inter_company/models/stock_picking.py

```diff
@@ -205,3 +197,25 @@ def _notify_picking_problem(self, purchase):
                 or SUPERUSER_ID,
             ),
         )
+
+    def _is_intercompany_reception(self):
+        """
+        Check if the picking is an inter-company reception.
+        Consider the customer location(dropship case) to avoid
+        explicitly adding a dependency on the stock_dropshipping module.
+        :return: bool
+        """
+        return (
+            self.location_dest_id.usage in ["internal", "customer"]
+            and self.purchase_id.sudo().intercompany_sale_order_id
+        )
```

Comment:
```suggestion
    def _is_intercompany_reception(self):
        """
        Check if the picking is an inter-company reception.
        :return: bool
        """
        return (
            self.location_id.usage == "supplier"
            and self.purchase_id.sudo().intercompany_sale_order_id
        )
```

Can't we check source location instead of destination for reception ? This would cover dropship ones too...

Reply:
Done